akka-stream/src/main/scala/akka/stream/impl/fusing/GraphStages.scala

```diff
@@ -306,6 +306,67 @@ object GraphStages {
     override def toString: String = s"SingleSource($elem)"
   }
 
+  final class FutureFlattenSource[T, M](
+    val future: Future[Graph[SourceShape[T], M]])
+    extends GraphStageWithMaterializedValue[SourceShape[T], Future[M]] {
+
+    ReactiveStreamsCompliance.requireNonNullElement(future)
+
+    val out = Outlet[T]("futureFlatten.out")
+    val shape = SourceShape(out)
+
+    override def initialAttributes = DefaultAttributes.futureSource // TODO
```

Comment:
That is handled by the framework automagically. You don't need to change anything here, this is how other stages work.